android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java

```diff
@@ -188,26 +183,6 @@ public void onCheckedChanged(
                 Utils.setBitmapResource(mExecutor, mHandler, context,
                         walletListItemModel.getIconPath(), walletListItemModel.getIcon(),
                         holder.iconImg, null, true);
-            } else {
-                PortfolioModel.NftDataModel nftDataModel = walletListItemModel.getNftDataModel();
-                if (walletListItemModel.hasNftImageLink()
-                        && ImageLoader.isSupported(nftDataModel.nftMetadata.mImageUrl)) {
-                    String url = nftDataModel.nftMetadata.mImageUrl;
-                    ImageLoader.downloadImage(url, Glide.with(context), false,
-                            WalletConstants.RECT_ROUNDED_CORNERS_DP, holder.iconImg, null);
-                } else {
-                    Utils.setBlockiesBitmapCustomAsset(mExecutor, mHandler, holder.iconImg,
-                            walletListItemModel.getBlockchainToken().contractAddress,
-                            walletListItemModel.getBlockchainToken().symbol,
-                            context.getResources().getDisplayMetrics().density, null, context,
-                            false, (float) 0.9, true);
-                }
-                if (mType == AdapterType.EDIT_VISIBLE_ASSETS_LIST) {
-                    onWalletListItemClick.onMaybeShowTrashButton(
-                            walletListItemModel, holder.iconTrash);
-                    holder.iconTrash.setOnClickListener(
-                            v -> onWalletListItemClick.onTrashIconClick(walletListItemModel));
-                }
             }
         } else {
```

Comment:
WalletCoinAdapter is now never instantiated with these types:
```
VISIBLE_ASSETS_LIST
EDIT_VISIBLE_ASSETS_LIST
```

Reply:
The entire implementation of this adapter was full of old refuse code, unneeded checks and incredibly hard to follow. I've performed a heavy refactor, renamed into a more appropriate name and pruned all the parts that do not make sense anymore. Very tedious (and time-consuming) work, but now it's much more readable. Let me know if you have more suggestions.